### PR TITLE
instanceof changed to recent syntax

### PR DESCRIPTION
**Pull request by dukbong**
instanceof changed to recent syntax

### CODE REVIEW

File: core/src/main/java/org/springframework/security/authentication/AbstractAuthenticationToken.java [hunk 1]
```diff
@@ -112,8 +112,8 @@ public void eraseCredentials() {
 	}
 
 	private void eraseSecret(Object secret) {
-		if (secret instanceof CredentialsContainer) {
-			((CredentialsContainer) secret).eraseCredentials();
+		if (secret instanceof CredentialsContainer secret) {
```
Comment (marcusdacoregio) [inline core/src/main/java/org/springframework/security/authentication/AbstractAuthenticationToken.java:115]: The variable name `secret` is already defined in the scope, can you please rename it to maybe `container`?

**Review by marcusdacoregio — CHANGES_REQUESTED** [review]
Thanks @dukbong, I've left a feedback inline.

**Comment by marcusdacoregio** [comment]
In preparation to merge, can you please squash your commits?

The single commit message could be: `Use instanceof Pattern Matching `

**Comment by dukbong** [comment]
> In preparation to merge, can you please squash your commits?
> 
> The single commit message could be: `Use instanceof Pattern Matching `

I'm sorry. @marcusdacoregio 

My mistake caused the PR to be removed.

I will submit a new PR.

**Pull request closed without merging**